src/mlpack/core/tree/greedy_single_tree_traverser_impl.hpp

```diff
@@ -36,13 +37,34 @@ void GreedySingleTreeTraverser<TreeType, RuleType>::Traverse(
   for (size_t i = 0; i < referenceNode.NumPoints(); ++i)
     rule.BaseCase(queryIndex, referenceNode.Point(i));
 
+  size_t bestChild = rule.GetBestChild(queryIndex, referenceNode);
+  size_t numDescendants;
+
+  // Check that referencenode is not a leaf node while calculating number of
+  // descendants of it's best child.
+  if (!referenceNode.IsLeaf())
+    numDescendants = referenceNode.Child(bestChild).NumDescendants();
+  else
+    numDescendants = referenceNode.NumPoints();
```

Comment:
You can move the whole definition of `numDescendants` into the `if (!referenceNode.IsLeaf())` block, and then you won't need the `if`/`else` here---you can just set

```
numDescendants = referenceNode.Child(bestChild).NumDescendants();
```

Reply:
I wrote about this in a comment, may be you missed it as it went down with "show outdated" :)
Here it is :
> We need to calculate numDescendant with !referenceNode.IsLeaf() condition imposed, otherwise on asking for child of leaf node it will give us segmentation fault. This was the silly error that I mentioned about in #1222 comment. It cause me a lot of trouble.
Please let me know if you mean the same thing.